### PR TITLE
Caplin: blob storage

### PR DESCRIPTION
**Pull request by bayram98**
This PR is to include `BlobSidecar`, `BlobIdentifier` data structure & storage. 

List of changes:
- Add `BlobSidecar` and `BlobIdentifier`
- SSZ (Simple Serialize) Marshaling/Unmarshaling
- Make it compatible with cl spectests. 
- Blob Storage `Write`,  `Read`, and `retrieveBlobsAndProofs(WIP)` 
- Simple unit tests

### CODE REVIEW

File: cl/persistence/blob_storage/blob.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	err = ssz_snappy.EncodeAndWrite(w, blobSidecar)
+	if err != nil {
+		return err
+	}
+	processBlock(a, blobSidecar, blockRoot)
+	return nil
+}
+
+// BlobSidecarReader retrieves a single BlobSidecar by its root
+func (a *aferoBlobStorage) BlobReader(blockRoot [32]byte) (*cltypes.BlobSidecar, error) {
```
Comment (Giulio2002) [inline cl/persistence/blob_storage/blob.go:76]: Rename to ReadBlobsByBlockRoot, also this should return multiples

File: cl/persistence/blob_storage/blob.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	buffer := bPool.Get().(*buffer.Buffer)
+	defer bPool.Put(buffer)
+	buffer.Reset()
+
+	fmt.Fprintf(buffer, "%d/%x.sz", slot/subDivisionFolderSize, root)
+	split := strings.Split(buffer.String(), "/")
+	return split[0], buffer.String()
+}
+
+func (a *aferoBlobStorage) BlobWriter(blobSidecar *cltypes.BlobSidecar) error {
```
Comment (Giulio2002) [inline cl/persistence/blob_storage/blob.go:54]: Rename to WriteBlob

**Review by Giulio2002 — CHANGES_REQUESTED** [review]

**Comment by Giulio2002** [comment]
One thing: as we discussed there are multiple blobs per block root

File: cl/persistence/blob_storage/blob.go (hunk 2 above)
Comment (bayram98) [reply cl/persistence/blob_storage/blob.go]: Fixed

**Review by Giulio2002 — APPROVED** [review]

**Pull request merged**